### ISSUE TITLE
queryRecord doesn't seem to return a fullly hydrated model

### ISSUE THREAD

**Issue by samselikoff** [emberjs/data#4227]
Working on a brand new 2.4 Ember app and I have the following route:

```js
import Ember from "ember";

export default Ember.Route.extend({

  model(params) {
    return this.store.queryRecord('video', {
      slug: params.slug
    });
  },

  afterModel(model) {
    debugger;
  }

});
```

`model.toString()` suggests I indeed have the model, but `model.get('title')` in `afterModel` (as well as `{{model.title}}` in my template) is blank. However, if I look in the Ember Inspector I see the data is there:

![image](https://cloud.githubusercontent.com/assets/2922250/13729126/f51b1cae-e903-11e5-8126-5f19a4718e22.png)

If I change the `model` hook to

```js
model(params) {
  return this.store.query('video', {
    slug: params.slug
  }).then(models => models.get('firstObject'));
}
```

then everything works.

The bug is there on `master` as well.

Here's a JSBIN: http://emberjs.jsbin.com/dajisaqufe/edit?html,js,console,output

**Comment by sly7-7** [comment]
I'm not sure, but I think `queryRecord`expects a single record to be returned from the server. 
Maybe I'm missing something, but http://emberjs.jsbin.com/zagohifevu/edit?html,js,console,output seems to work as expected

**Comment by pangratz** [comment]
If I recall correctly there was once the implementation where the first item is used, if an array is returned (also see last comment by @bcardarella [here](https://github.com/emberjs/data/issues/3977#issuecomment-163163862)). It looks like this is [no more the case](https://github.com/emberjs/data/blob/v2.4.0/addon/serializers/json.js#L450-L454)...

**Comment by samselikoff** [comment]
Ah I see, ok closing for now. Thanks!

**Comment by pangratz** [comment]
@samselikoff what adapter are you using? I assume `JSON-API`?

**Comment by samselikoff** [comment]
Yep

**Comment by pangratz** [comment]
For clarification of my comment above:

> If I recall correctly there was once the implementation where the first item is used, if an array is returned 

This is currently the case for the `rest` adapter, which returns the first object when `queryRecord` returns an array.

---

#4300 addressed some issues with `queryRecord` and the outcome is that `store.queryRecord` should only be used, when a single record is returned and the id is not known beforehand. For all other cases `store.query` in combination with returning the first object in the array is encouraged. Assertions added in #4300 should now inform you when an array is returned for `queryRecord`.

I am going to close this issue as I think this has been addressed in #4300. @samselikoff feel free to re-open if you disagree, so we can discuss further!
